packages/file_selector/file_selector_macos/pigeons/messages.dart

```diff
@@ -50,10 +47,10 @@ class SavePanelOptions {
 /// These correspond to NSOpenPanel properties.
 class OpenPanelOptions extends SavePanelOptions {
   const OpenPanelOptions({
-    this.allowsMultipleSelection = false,
-    this.canChooseDirectories = false,
-    this.canChooseFiles = true,
-    this.baseOptions = const SavePanelOptions(),
+    required this.allowsMultipleSelection,
+    required this.canChooseDirectories,
+    required this.canChooseFiles,
+    required this.baseOptions,
```

Comment:
Since this change is non-obvious: Pigeon used to basically discard the constructor, so none of this mattered. With the update:
- some code failed analysis because it was passing default values, and I decided I preferred having the call site across the Pigeon boundary be very explicit about the settings in each method, and
- the `const SavePanelOptions()` didn't work because const-ness isn't preserved into the Pigeon-generated Dart code.

Reply:
is const-ness something we'd like?

Reply:
Could be worth considering, since having data classes be unmodifiable is often useful. It would add some complexity to native generators though since we'd need to make setters conditionally generated.